### PR TITLE
fix facebook auth for new v2 api

### PR DESCRIPTION
**Pull request by cjekel**
Fixes facebook authentication for new v2 api.

Changes:
- new auth url
- json headers are no longer needed
- send post directly to auth url without headers

Test results:
```
Name                        Stmts   Miss  Cover
-----------------------------------------------
pynder/__init__.py              1      0   100%
pynder/api.py                  97     41    58%
pynder/constants.py            14      0   100%
pynder/errors.py               10      0   100%
pynder/models/__init__.py       5      0   100%
pynder/models/base.py           2      0   100%
pynder/models/friend.py        25     17    32%
pynder/models/me.py            69     24    65%
pynder/models/message.py       35     23    34%
pynder/models/user.py          99     62    37%
pynder/session.py              49     19    61%
-----------------------------------------------
TOTAL                         406    186    54%
----------------------------------------------------------------------
Ran 3 tests in 0.861s

OK
```

Closes:
https://github.com/charliewolf/pynder/issues/208

### CODE REVIEW

File: pynder/api.py [hunk 1]
```diff
@@ -22,21 +21,21 @@ def _full_url(self, url):
...
+        self._token = result['api_token']
+        self._session.headers.update({"X-Auth-Token": str(result['api_token'])})
         return result
 
     def _request(self, method, url, data={}):
         if not hasattr(self, '_token'):
             raise errors.InitializationError
         result = self._session.request(method, self._full_url(
-            url), json=data, proxies=self._proxies)
+            url), data=data, proxies=self._proxies)
```
Comment (cjekel) [inline pynder/api.py]: Can someone comment if this line needed to be changed?

File: pynder/constants.py [hunk 2]
```diff
@@ -3,17 +3,18 @@
...
-HEADERS = {
-    "Content-Type": "application/json; charset=utf-8",
-    "User-Agent": USER_AGENT,
-    "Host": API_BASE,
-    "os_version": "1935",
-    "app-version": "371",
-    "platform": "android",  # XXX with ios we run in an error
-    "Accept-Encoding": "gzip"
-}
+# No longer needed! 2019-04-29 ??????
```
Comment (cjekel) [inline pynder/constants.py]: While this worked... it's a bad idea...

**Comment by cjekel** [comment]
It's probably a bad idea to auth w/o headers so sorry for the trouble this may have caused...

**Comment by charlesduponpon** [comment]
@cjekel did you find a solution? I tried your this PR's branch with tindetheus but I still got the same JSON decode error. By the way, tindetheus looks super cool, I can't wait to try it!

**Comment by cjekel** [comment]
@charlesduponpon  I couldn't find headers that worked with the v2 facebook authentication. 

https://github.com/charliewolf/pynder/pull/210/commits/5898e8b7e0df1b5ed5e87597c5ffcd31c352f5a1 was a version of pynder without headers, where I was able to login and use pynder briefly. I was able to like/dislike profiles, but I'm not sure if I was able to update my profile. 

I then though that using https://github.com/charliewolf/pynder/pull/210/commits/5898e8b7e0df1b5ed5e87597c5ffcd31c352f5a1 was a bad idea, since you are sending a user agent of Python requests to the tinder api (rather than a spoofed phone). So this would have been a bad temporary solution, rather than a long term fix.

**Comment by charlesduponpon** [comment]
I see. Thanks for the explanation. I'm well versed into machine learning but not so much about networking. Not sure I can help with this problem. I hope it gets solved soon.

**Pull request closed without merging**